### PR TITLE
Rename applications on agent menu

### LINKED ISSUE #5893: Update the new sidebar menu design
**Description**

A new sidebar menu design was defined in a recent meeting.

We got a new sidebar menu defined in a recent meeting. There are some changes that we should apply to the previous design.

# Requirements update

| Category | Name | ID | Section in monolith plugin | Changes |
|---|---|---|---|---|
| Home | Overview | wz-home | Modules directory | This is the home (the home icon) |
| Endpoint security | Security Configuration Assessment | security-configuration-assessment | Modules > Security Configuration Assessment | RENAME Configuration assessment |
| Endpoint security | Policy monitoring | policy-monitoring | Modules > Policy monitoring | RENAME to Malware detection  (this is rootcheck now) |
| Endpoint security | Integrity monitoring | integrity-monitoring | Modules > Integrity monitoring | RENAME to File integrity monitoring |
| Endpoint security | System auditing | system-auditing | Modules > System auditing | DELETE |
| Endpoint security | OpenSCAP | openscap | Modules > OpenSCAP | DELETE |
| Endpoint security | CIS-CAT | ciscat | Modules > CIS-CAT | DELETE |
| Threat intelligence | Security events | security-events | Modules > Security events | RENAME Threat hunting |
| Threat intelligence | Vulnerabilities | vulnerabilities | Modules > Vulnerabilities | RENAME to Vulnerability detection |
| Threat intelligence | MITRE ATT&CK | mitre-attack | Modules > MITRE ATT&CK | |
| Threat intelligence | Virustotal | virustotal | Modules > Virustotal | |
| Security operations | PCI DSS | pci-dss | Modules > PCI DSS | |
| Security operations | GDPR | gdpr | Modules > GDPR | |
| Security operations | HIPAA | hipaa | Modules > HIPAA | |
| Security operations | NIST 800-53 | nist-800-53 | Modules > NIST 800-53 | |
| Security operations | TSC | tsc | Modules > TSC | |
| Security operations | IT Hygiene | it-hygiene | Agents > {agent_id} -> Inventory |  |
| Security operations | Osquery | osquery | Modules > Osquery | DELETE |
| Cloud security | AWS | amazon-web-services | Modules > AWS | |
| Cloud security | Google Cloud | google-cloud | Modules > Google Cloud | |
| Cloud security | GitHub | github | Modules > GitHub | |
| Cloud security | Office365 | office365 | Modules > Office365 | |
| Cloud security | Docker listener | docker-listener | Modules > Docker listener | RENAME to Docker |
| Explore | Discover | - | - | Renamed from "Openseach dashboards" |
| Explore | Dashboards | - | - | |
| Explore | Visualize | - | - | |
| Explore | Reporting | - | - | Comes from "Opensearch plugins"  |
| Explore | Alerting | - | - | |
| Explore | Maps | - | - | |
| Explore | Notifications | - | - | |
| Server management | Endpoints summary | endpoints-summary | Agents | This comes from Endpoint security |
| Server management | Groups | groups | Management > Groups | RENAME to Endpoint groups |
| Server management | Rules | rules | Management > Rules | |
| Server management | Decoders | decoders | Management > Decoders | |
| Server management | CDB lists | cdb-lists | Management > CDB lists | |
| Server management | Status | server-status | Management > Status | |
| Server management | Cluster | cluster | Management > Cluster | |
| Server management | Statistics | statistics | Management > Statistics | |
| Server management | Logs | logs | Management > Logs | |
| Server management | Settings | settings | Management > Configuration | |
| Server management | API console | api-console | Tools > API console | RENAME Dev tools |
| Server management | Ruleset test | ruleset-test | Tools > Ruleset test | |
| Server management | RBAC | rbac | Security | RENAME to Security |
| Indexer/dashboard Management | Reporting | reporting | Management > Reporting | Comes from "server management" | REMOVE and use Explore reporting highjacking the download capability of the standard plugin |
| Indexer/dashboard Management | Server API | server-api | Settings > API configuration | |
| Indexer/dashboard Management | Modules | modules | Settings > Modules | DELETE |
| Indexer/dashboard Management | Server data | server-data | Settings > Sample data | |
| Indexer/dashboard Management | Configuration | configuration | Settings > Configuration | RENAME to Wazuh plugin settings |
| Indexer/dashboard Management | Logs | app-logs | Settings > Logs | RENAME to Wazuh plugin logs |
| Indexer/dashboard Management | About | about | Settings > About | RENAME to Wazuh plugin about |
| Indexer/dashboard Management | Dev tools | - | - | Comes from management |
| Indexer/dashboard Management | Stack management | - | - | Comes from management |
| Indexer/dashboard Management | Security | - | - | -> Comes from Opensearch plugins, nice to move it to indexer./dashaboard management  |
| Indexer/dashboard Management| Index management | - | - | |
| Indexer/dashboard Management | Snapshot management | - | - | |

Tasks
- [x] Update the sidebar with the new design
- [x] Adapt the agent menu to the applications related to the agent
- [ ] Move the built-in applications to the expected category
- [x] Enhance the order of applications

### PR DESCRIPTION
**Pull request by Desvelao**
### Description
This pull request adapts the agent menu to display the applications related to agents.

Changes: 
- Refactor agent menu to use the applications and categories
- Simplify the data saved in the local storage to keep the pinned applications
- Remove the usage of extensions configuration
- Change the data schema of the pinned applications
- Rename the session storage key to `wz-menu-agent-apps-pinned` where the pinned applications are stored
- Fix a problem with the agent menu header when the side menu is docked

### Issues Resolved
#5893 

### Evidence
image
image
image
image

### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome |
| --- |  --- |
| With a fresh browser, ensure the Threat hunting, File integrity monitoring, Configuration assessment, Vulnerability detection and MITRE ATT&CK applications are displayed as the default ones in the agent welcome view | :black_circle: |
| Clicking on the More button of agent welcome, should display all the applications | :black_circle: |
| Pinning an application from the More/Applications menu should display that application out of the More menu | :black_circle: |
| Unpinning an application from the More/Applications menu should not display that application out of the More menu | :black_circle: |
| Using a small window width, the More button on the agent welcome should display as Applications | :black_circle: |
| Modify the pinned applications from the agent welcome, close the browser, open it and navigate the the agent welcome view. The pinned applications should be kept | :black_circle: |
| The agent's navbar should display correctly | :black_circle: |
| The agent's navbar should display correctly with the dock menu. | :black_circle: |
| Inventory data, statistics and configuration must be displayed correctly | :black_circle: |

**Details**
<details>
<summary>:black_circle: With a fresh browser, ensure the Threat hunting, File integrity monitoring, Configuration assessment, Vulnerability detection and MITRE ATT&CK applications are displayed as the default ones in the agent welcome view</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Clicking on the More button of agent welcome, should display all the applications</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Pinning an application from the More/Applications menu should display that application out of the More menu</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Unpinning an application from the More/Applications menu should not display that application out of the More menu</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Using a small window width, the More button on the agent welcome should display as Applications</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Modify the pinned applications from the agent welcome, close the browser, open it and navigate the the agent welcome view. The pinned applications should be kept</summary>

Chrome - :black_circle:

</details>
<details>
<summary>:black_circle: The agent's navbar should display correctly</summary>

Chrome - :black_circle:

</details>
<details>
<summary>:black_circle: The agent's navbar should display correctly with the dock menu</summary>

Chrome - :black_circle:

</details>
<details>
<summary>:black_circle: Inventory data, statistics and configuration must be displayed correctly</summary>

Chrome - :black_circle:

</details>

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: plugins/main/public/utils/applications.ts [hunk 1]
```diff
@@ -533,7 +562,7 @@ const settings = {
   }),
   euiIconType: 'indexRollupApp',
   showInOverviewApp: false,
-  redirectTo: () => '/manager/?tab=configuration',
```
Comment (yenienserrano) [inline plugins/main/public/utils/applications.ts:536]: This should be in place for the redirect to work

File: plugins/main/public/utils/applications.ts [hunk 2]
```diff
@@ -618,7 +652,7 @@ const wazuhPluginSettings = {
   }),
   euiIconType: 'indexRollupApp',
   showInOverviewApp: false,
-  redirectTo: () => '/settings?tab=configuration',
```
Comment (yenienserrano) [inline plugins/main/public/utils/applications.ts:621]: This should be in place for the redirect to work

**Comment by yenienserrano** [comment]
### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome |
| --- |  --- |
| With a fresh browser, ensure the Threat hunting, File integrity monitoring, Configuration assessment, Vulnerability detection and MITRE ATT&CK applications are displayed as the default ones in the agent welcome view | :black_circle: |
| Clicking on the More button of agent welcome, should display all the applications | :black_circle: |
| Pinning an application from the More/Applications menu should display that application out of the More menu | :black_circle: |
| Unpinning an application from the More/Applications menu should not display that application out of the More menu | :black_circle: |
| Using a small window width, the More button on the agent welcome should display as Applications | :black_circle: |
| Modify the pinned applications from the agent welcome, close the browser, open it and navigate the the agent welcome view. The pinned applications should be kept | :black_circle: |

**Details**
<details>
<summary>:black_circle: With a fresh browser, ensure the Threat hunting, File integrity monitoring, Configuration assessment, Vulnerability detection and MITRE ATT&CK applications are displayed as the default ones in the agent welcome view</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Clicking on the More button of agent welcome, should display all the applications</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Pinning an application from the More/Applications menu should display that application out of the More menu</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Unpinning an application from the More/Applications menu should not display that application out of the More menu</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Using a small window width, the More button on the agent welcome should display as Applications</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Modify the pinned applications from the agent welcome, close the browser, open it and navigate the the agent welcome view. The pinned applications should be kept</summary>

Chrome - :black_circle:

</details>

File: plugins/main/public/utils/applications.ts (hunk 2 above)
Comment (yenienserrano) [reply plugins/main/public/utils/applications.ts:621]: Done https://github.com/wazuh/wazuh-dashboard-plugins/pull/5997/commits/1fa1cd661038981db715e06ad0a9597ce1e13bda

File: plugins/main/public/utils/applications.ts (hunk 1 above)
Comment (yenienserrano) [reply plugins/main/public/utils/applications.ts:536]: Done https://github.com/wazuh/wazuh-dashboard-plugins/pull/5997/commits/1fa1cd661038981db715e06ad0a9597ce1e13bda

**Comment by Desvelao** [comment]
Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

> Partial tests
| Test | Chrome |
| --- |  --- |
| The agent's navbar should display correctly | :green_circle: |
| The agent's navbar should display correctly with the dock menu. | :green_circle: |
| Inventory data, statistics and configuration must be displayed correctly | :green_circle: |


**Details**
<details>
<summary>:green_circle: The agent's navbar should display correctly</summary>

Chrome - :green_circle:
image

</details>
<details>
<summary>:green_circle: The agent's navbar should display correctly with the dock menu</summary>

Chrome - :green_circle:
image

</details>
<details>
<summary>:green_circle: Inventory data, statistics and configuration must be displayed correctly</summary>

Chrome - :green_circle:
image
image
image
image
image


</details>

**Review by yenienserrano — APPROVED** [review]
LGTM

**Pull request merged**